ui/src/views/Secrets/ViewSecrets/index.jsx

```diff
@@ -51,6 +64,10 @@ export default class ViewSecrets extends Component {
         ? { name: { $regex: escapeStringRegexp(secretSearch), $options: 'i' } }
         : null,
     });
+
+    this.props.history.push(
+      `/secrets${secretSearch ? `?search=${secretSearch}` : ''}`
```

Comment:
Let's encode the search term before pushing it to history similar to how it's done in https://github.com/taskcluster/taskcluster/pull/2660/commits/bd9db1a9905a204a816379468d401394e6ef740c.

Reply:
On a second thought, there's no need to encode if we use the `stringify` function from the `qs` package. https://github.com/taskcluster/taskcluster/blob/978b8dc93e54d5ec34cfbd06cad6c88796f53f5f/ui/src/views/Hooks/ListHooks/index.jsx#L36-L45